Maple2.Server.Game/Manager/Field/FieldManager.State.cs

```diff
@@ -365,6 +365,24 @@ public partial class FieldManager {
         Broadcast(FieldPropertyPacket.Background(background));
     }
 
+    private void SetBonusMapPortal(IList<MapMetadata> bonusMaps, Ms2RegionSpawn spawn) {
+        // Spawn a hat within a random range of 5 min to 8 hours
+        int delay = Random.Shared.Next(1, 97) * (int) TimeSpan.FromMinutes(5).TotalMilliseconds;
+        var portal = new Portal(NextLocalId(), bonusMaps[Random.Shared.Next(bonusMaps.Count)].Id,  -1, PortalType.Event, PortalActionType.Interact, spawn.Position, spawn.Rotation,
+            new Vector3(200, 200, 250), 0, true, false, true);
+        FieldPortal fieldPortal = SpawnPortal(portal);
+        fieldPortal.Model = Metadata.Property.Continent switch {
```

Comment:
you think there's an xblock we should be parsing for this data?

Reply:
I looked at the xblocks and no dice. I think these are within Model.m2d.

Reply:
yea i guess xblock are only static models, and since these are server-sided only it doesnt show up. Nexon probably has something else that includes this data.